code/modules/antagonists/creep/creep.dm

```diff
@@ -0,0 +1,174 @@
+				objectives += polaroid
+
+	objectives += kill//finally add the assassinate last, because you'd have to complete it last to greentext.
+	for(var/datum/objective/O in objectives)
+		O.update_explanation_text()
+
+/datum/antagonist/creep/roundend_report_header()
+	return 	"<span class='header'>Someone became a creep!</span><br>"
+
+/datum/antagonist/creep/roundend_report()
+	var/list/report = list()
+
+	if(!owner)
+		CRASH("antagonist datum without owner")
+
+	report += "<b>[printplayer(owner)]</b>"
+
+	var/objectives_complete = TRUE
+	if(objectives.len)
+		report += printobjectives(objectives)
+		for(var/datum/objective/objective in objectives)
+			if(!objective.check_completion())
+				objectives_complete = FALSE
+				break
+	if(trauma)
+		if(trauma.total_time_creeping > 0)
+			report += "<span class='greentext'>The [name] spent a total of [DisplayTimeText(trauma.total_time_creeping)] being near [trauma.obsession]!</span>"
+		else
+			report += "<span class='redtext'>The [name] did not go near their obsession the entire round! That's extremely impressive, but you are a shit [name]!</span>"
+	else
+		report += "<span class='greentext'>The [name] had no trauma attached to their antagonist ways! Either it bugged out or an admin incorrectly gave this good samaritan antag and it broke! You might as well show yourself!!</span>"
+
+	if(objectives.len == 0 || objectives_complete)
+		report += "<span class='greentext big'>The [name] was successful!</span>"
+	else
+		report += "<span class='redtext big'>The [name] has failed!</span>"
+
+	return report.Join("<br>")
+
+//////////////////////////////////////////////////
+///CREEPY objectives (few chosen per obsession)///
+//////////////////////////////////////////////////
+
+/datum/objective/assassinate/creep //just a creepy version of assassinate
+
+/datum/objective/assassinate/creep/update_explanation_text()
+	..()
+	if(target && target.current)
+		explanation_text = "Murder [target.name], the [!target_role_type ? target.assigned_role : target.special_role]."
+	else
+		explanation_text = "Free Objective"
+
+/datum/objective/spendtime //spend some time around someone, handled by the creep trauma since that ticks
+	var/timer = 3000 //5 minutes
+	var/freeobj = FALSE //if it can't find the trauma to attach it to.
+
+/datum/objective/spendtime/New()
+	var/datum/antagonist/creep/creeper = owner.has_antag_datum(/datum/antagonist/creep)
+	if(!creeper)
+		return
+	if(creeper.trauma)
+		creeper.trauma.attachedcreepobj = src
+	else
+		freeobj = TRUE
+
+/datum/objective/spendtime/update_explanation_text()
+	if(timer == 3000)//just so admins can mess with it
+		timer += pick(-1200, -600, 600, 1200)
+	if(target && target.current && !freeobj)
+		explanation_text = "Spend [DisplayTimeText(timer)] around [target.name] while they're alive."
+	else
+		explanation_text = "Free Objective"
+
+/datum/objective/spendtime/check_completion()
+	return timer <= 0 || freeobj
+
+/*
+/datum/objective/hug
```

Comment:
hey, any good way to make this work by the way?

Reply:
Send a signal when a player is hugged

Reply:
good idea